server/graphql/general/src/queries/insurances.rs

```diff
@@ -0,0 +1,199 @@
+use async_graphql::{Context, Enum, InputObject, Object, Result, SimpleObject, Union};
+use chrono::NaiveDate;
+use graphql_core::{
+    generic_filters::EqualFilterStringInput,
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use repository::{
+    name_insurance_join_row::{
+        InsurancePolicyType, NameInsuranceJoinFilter, NameInsuranceJoinRow, NameInsuranceJoinSort,
+        NameInsuranceJoinSortField,
+    },
+    EqualFilter,
+};
+use serde::Serialize;
+use service::auth::{Resource, ResourceAccessRequest};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(rename_items = "camelCase")]
+pub enum InsuranceSortFieldInput {
+    ExpiryDate,
+    IsActive,
+}
+
+#[derive(InputObject)]
+pub struct InsuranceSortInput {
+    /// Sort query result by `key`
+    key: InsuranceSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct InsuranceFilterInput {
+    pub insurance_provider_id: Option<EqualFilterStringInput>,
+}
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum InsurancePolicyNodeType {
+    Personal,
+    Business,
+}
+
+impl InsurancePolicyNodeType {
+    pub fn from_domain(policy_type: &InsurancePolicyType) -> InsurancePolicyNodeType {
+        use InsurancePolicyType::*;
+        match policy_type {
+            Personal => InsurancePolicyNodeType::Personal,
+            Business => InsurancePolicyNodeType::Business,
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct InsuranceNode {
+    insurance: NameInsuranceJoinRow,
+}
+
+#[Object]
+impl InsuranceNode {
+    pub async fn id(&self) -> &str {
+        &self.row().id
+    }
+
+    pub async fn name_link_id(&self) -> &str {
+        &self.row().name_link_id
+    }
+
+    pub async fn insurance_provider_id(&self) -> &str {
+        &self.row().insurance_provider_id
+    }
+
+    pub async fn policy_number_person(&self) -> &Option<String> {
+        &self.row().policy_number_person
+    }
+
+    pub async fn policy_number_family(&self) -> &Option<String> {
+        &self.row().policy_number_family
+    }
+
+    pub async fn policy_number(&self) -> &String {
+        &self.row().policy_number
+    }
+
+    pub async fn policy_type(&self) -> InsurancePolicyNodeType {
+        InsurancePolicyNodeType::from_domain(&self.row().policy_type)
+    }
+
+    pub async fn discount_percentage(&self) -> &f64 {
+        &self.row().discount_percentage
+    }
+
+    pub async fn expiry_date(&self) -> &NaiveDate {
+        &self.row().expiry_date
+    }
+
+    pub async fn is_active(&self) -> &bool {
+        &self.row().is_active
+    }
+
+    pub async fn entered_by_id(&self) -> &Option<String> {
```

Comment:
I'm not sure this is being populated by mSupply? Maybe best to leave it out for now, we're not displaying it right?